lib/web/command_utils.go

```diff
@@ -130,3 +137,63 @@ func (s *syncRWWSConn) ReadMessage() (messageType int, p []byte, err error) {
 	defer s.rmtx.Unlock()
 	return s.WSConn.ReadMessage()
 }
+
+func newBufferedPayloadWriter(pw *payloadWriter, buffer *summaryBuffer) *bufferedPayloadWriter {
+	return &bufferedPayloadWriter{
+		payloadWriter: pw,
+		buffer:        buffer,
+	}
+}
+
+type bufferedPayloadWriter struct {
+	*payloadWriter
+	buffer *summaryBuffer
+}
+
+func (bp *bufferedPayloadWriter) Write(data []byte) (int, error) {
+	bp.buffer.Write(bp.nodeID, data)
+	return bp.payloadWriter.Write(data)
+}
+
+func newSummaryBuffer(capacity int) *summaryBuffer {
+	return &summaryBuffer{
+		buffer:            make(map[string][]byte),
+		remainingCapacity: capacity,
+		invalid:           false,
+		mutex:             sync.Mutex{},
+	}
+}
+
+type summaryBuffer struct {
+	buffer            map[string][]byte
+	remainingCapacity int
+	invalid           bool
+	mutex             sync.Mutex
+}
+
+func (b *summaryBuffer) Write(node string, data []byte) {
+	b.mutex.Lock()
+	defer b.mutex.Unlock()
+	if b.invalid {
+		return
+	}
+	if len(data) > b.remainingCapacity {
+		// We're out of capacity, not all content will be written to the buffer
+		// it should not be used anymore
+		b.invalid = true
+		return
+	}
+	b.buffer[node] = append(b.buffer[node], data...)
+	b.remainingCapacity -= len(data)
+}
+
+func (b *summaryBuffer) Export() map[string][]byte {
```

Comment:
If return `map[string][]byte, bool` similar to Go's `map` to distinguish between no output and overflow.